StortSpelProjekt/Engine/src/Renderer/Texture/Texture.cpp

```diff
@@ -0,0 +1,46 @@
+#include "stdafx.h"
+#include "Texture.h"
+
+#include "../GPUMemory/Resource.h"
+#include "../CommandInterface.h"
+#include "../GPUMemory/ShaderResourceView.h"
+#include "../DescriptorHeap.h"
+
+Texture::Texture()
+{
+
+}
+
+Texture::~Texture()
+{
+	// TODO: Cleanup deconstructor
+	if (m_pImageData != nullptr)
+	{
+		delete m_pImageData;
+	}
+
+	if (m_pDefaultResource != nullptr)
+	{
+		delete m_pDefaultResource;
+	}
+
+	if (m_pUploadResource != nullptr)
+	{
+		delete m_pUploadResource;
+	}
+
+	if (m_pSRV != nullptr)
+	{
+		delete m_pSRV;
+	}
```

Comment:
Clean this up

Reply:
Just removed comment, comment was there for when I had delete m_DdsData and another variable I cleaned up.
The deconstructor looks like this since we allocate these variables in the Init() function which we do not know if it has been called and allocated.